backends/sycl-ref/ceed-sycl-ref-operator.sycl.cpp

```diff
@@ -992,63 +1000,81 @@ static int CeedSingleOperatorAssembleSetup_Sycl(CeedOperator op) {
+  // However, for the purposes of loading the B matrices, it will be treated as one mode, and we will load/copy the entire gradient matrix at once,
   // so num_B_in_mats_to_load will be incremented by 1.
   for (CeedInt i = 0; i < num_input_fields; i++) {
-    CeedEvalMode eval_mode;
-    CeedVector   vec;
+    CeedVector vec;
 
     CeedCallBackend(CeedOperatorFieldGetVector(input_fields[i], &vec));
     if (vec == CEED_VECTOR_ACTIVE) {
-      CeedCallBackend(CeedOperatorFieldGetBasis(input_fields[i], &basis_in));
+      CeedEvalMode        eval_mode;
+      CeedElemRestriction elem_rstr;
+      CeedBasis           basis;
+
+      CeedCallBackend(CeedOperatorFieldGetBasis(input_fields[i], &basis));
+      CeedCheck(!basis_in || basis_in == basis, ceed, CEED_ERROR_BACKEND, "Backend does not implement operator assembly with multiple active bases");
+      if (!basis_in) CeedCallBackend(CeedBasisReferenceCopy(basis, &basis_in));
+      CeedCallBackend(CeedBasisDestroy(&basis));
       CeedCallBackend(CeedBasisGetDimension(basis_in, &dim));
       CeedCallBackend(CeedBasisGetNumQuadraturePoints(basis_in, &num_qpts));
-      CeedCallBackend(CeedOperatorFieldGetElemRestriction(input_fields[i], &rstr_in));
+      CeedCallBackend(CeedOperatorFieldGetElemRestriction(input_fields[i], &elem_rstr));
+      if (!rstr_in) CeedCallBackend(CeedElemRestrictionReferenceCopy(elem_rstr, &rstr_in));
+      CeedCallBackend(CeedElemRestrictionDestroy(&elem_rstr));
       CeedCallBackend(CeedElemRestrictionGetElementSize(rstr_in, &elem_size));
       CeedCallBackend(CeedQFunctionFieldGetEvalMode(qf_fields[i], &eval_mode));
       if (eval_mode != CEED_EVAL_NONE) {
         CeedCallBackend(CeedRealloc(num_B_in_mats_to_load + 1, &eval_mode_in));
         eval_mode_in[num_B_in_mats_to_load] = eval_mode;
         num_B_in_mats_to_load += 1;
         if (eval_mode == CEED_EVAL_GRAD) {
-          num_e_mode_in += dim;
+          num_eval_mode_in += dim;
           size_B_in += dim * elem_size * num_qpts;
         } else {
-          num_e_mode_in += 1;
+          num_eval_mode_in += 1;
           size_B_in += elem_size * num_qpts;
         }
       }
     }
+    CeedCallBackend(CeedVectorDestroy(&vec));
   }
 
   // Determine active output basis; basis_out and rstr_out only used if same as input, TODO
   CeedCallBackend(CeedQFunctionGetFields(qf, NULL, NULL, NULL, &qf_fields));
   for (CeedInt i = 0; i < num_output_fields; i++) {
-    CeedEvalMode eval_mode;
-    CeedVector   vec;
+    CeedVector vec;
 
     CeedCallBackend(CeedOperatorFieldGetVector(output_fields[i], &vec));
     if (vec == CEED_VECTOR_ACTIVE) {
-      CeedCallBackend(CeedOperatorFieldGetBasis(output_fields[i], &basis_out));
-      CeedCallBackend(CeedOperatorFieldGetElemRestriction(output_fields[i], &rstr_out));
-      CeedCheck(!rstr_out || rstr_out == rstr_in, ceed, CEED_ERROR_BACKEND, "Backend does not implement multi-field non-composite operator assembly");
```

Comment:
Is the change in check here deliberate?

Reply:
Obviously it aligns more with the CUDA and HIP backends, but just making sure we're at least mostly certain that the check isn't necessary anymore.

Reply:
The SYCL backend is a huge problem IMHO. During development, changes were made from the CUDA/HIP source that didn't have well captured reasons, and since then the CUDA/HIP source has changed, so its really tricky to make good changes here, especially with the limited access to a good development environment. I don't see a good fix in the near term

Reply:
@jedbrown Thoughts on getting Jeremy access to Sunspot? CI would be nice too, but these kind of dev changes would be easier in a shorter debug-loop.

Reply:
Giving me a way to build and test would be very helpful - I'd like to make a number of code quality fixes and transfer some CUDA and HIP improvements over

Reply:
I don't have authority to grant Sunspot access (cc: @KennethEJansen ). I would consider buying a (modestly priced) Intel GPU to put in Noether for dev/testing. Cursory web search hasn't been good at finding actual vendors and I don't know what pricing on the Max 1100 looks like.

Reply:
Yeah, one way or another it would be hugely helpful to the health of those backends to get me access to Sunspot or to get an Intel card somewhere ourselves, otherwise they run the risk of becoming like the OCCA backend